### PR TITLE
Completed test coverage for django.utils.dateformat and simplified some formats.

### PR DESCRIPTION
**Pull request by pope1ni**
I've left this split into separate commits to make it easier to follow.

### CODE REVIEW

**Comment by pope1ni** [comment]
@felixxm I've rebased this one.

Your https://github.com/django/django/pull/14602#issuecomment-875491043 mentioned that this PR might address use of `subTest()`, but that is for `django.utils.dateparse` and this is for `django.utils.dateformat`. Regardless, this is ready for review.

File: tests/utils_tests/test_dateformat.py [hunk 1]
```diff
@@ -105,6 +107,22 @@ def test_time_formats(self):
...
+        self.assertEqual(dateformat.format(my_birthday, 'A'), 'PM')
+        self.assertEqual(dateformat.format(my_birthday, 'f'), '10')
+        self.assertEqual(dateformat.format(my_birthday, 'g'), '10')
+        self.assertEqual(dateformat.format(my_birthday, 'G'), '22')
+        self.assertEqual(dateformat.format(my_birthday, 'h'), '10')
+        self.assertEqual(dateformat.format(my_birthday, 'H'), '22')
+        self.assertEqual(dateformat.format(my_birthday, 'i'), '00')
+        self.assertEqual(dateformat.format(my_birthday, 'P'), '10 p.m.')
+        self.assertEqual(dateformat.format(my_birthday, 's'), '00')
+        self.assertEqual(dateformat.format(my_birthday, 'u'), '000000')
```
Comment (felixxm) [inline tests/utils_tests/test_dateformat.py:124]: Can we use `subTest()`?

File: tests/utils_tests/test_dateformat.py [hunk 2]
```diff
@@ -67,36 +67,38 @@ def test_empty_format(self):
...
+        self.assertEqual(dateformat.time_format(morning, 'A'), 'AM')
+        self.assertEqual(dateformat.time_format(evening, 'A'), 'PM')
 
     def test_microsecond(self):
         # Regression test for #18951
         dt = datetime(2009, 5, 16, microsecond=123)
         self.assertEqual(dateformat.format(dt, 'u'), '000123')
 
     def test_date_formats(self):
+        # Specifiers 'I', 'r', and 'U' are covered in test_timezones().
```
Comment (felixxm) [inline tests/utils_tests/test_dateformat.py:84]: Can we use `subTest()`?

**Comment by pope1ni** [comment]
Thanks @felixxm. Updated.

**Comment by felixxm** [comment]
@pope1ni Thanks :+1: I reorganized commits.

**Pull request merged**